soroban-env-host/src/e2e_invoke.rs

```diff
@@ -892,11 +952,18 @@ fn build_storage_map_from_xdr_ledger_entries<T: AsRef<[u8]>, I: ExactSizeIterato
                 budget,
             )?;
             if ttl_entry.live_until_ledger_seq < ledger_num {
-                return Err(Error::from_type_and_code(
-                    ScErrorType::Storage,
-                    ScErrorCode::InternalError,
-                )
-                .into());
+                if !allow_expired_entries {
```

Comment:
Instead of introducing `allow_expired_entries`, could the logic be gated on enforcing vs recording mode?

Reply:
If you mean the feature guard, then no. If you mean just renaming this, that makes sense, renamed to `is_recording_mode`.
I also realized though that we only did this argument in recording mode, so I've simplified the enforcing path (which increases the overall amount of code, but seems safer).